Add count_include_pad to average_pool_gradient_op

xiaomengy opened the pull request:
Summary: Add count_include_pad to average_pool_gradient_op

Differential Revision: D13648339

BIT-silence commented:
Benchmark:
N = 128, C = 256, H = 128, W = 128
AveragePoolGrad with NCHW order, stride = 4, kernel = 4: 93.3ms -> 58.72ms
AveragePoolGrad with NHWC order, stride = 4, kernel = 4: 93.3ms -> 37.4ms (CuDNN 108.56ms)
MaxPoolGrad with NCHW order, stride = 2, kernel = 2: 91.9ms -> 53.12ms
MaxPoolGrad with NCHW order, stride = 2, kernel = 2: 92.3ms -> 32.3ms (CuDNN 269.6ms)

houseroad reviewed (approved):
Looks good to me. Please fix the hip test before ship it.

xiaomengy committed:
Add count_include_pad to average_pool_gradient_op (#15997)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/15997

Add count_include_pad to average_pool_gradient_op

Reviewed By: houseroad

Differential Revision: D13648339

fbshipit-source-id: 8f558889b950009d459ec05d18257e7c7d669ddd

BIT-silence commented on caffe2/python/operator_test/pooling_test.py:
```diff
@@ -335,6 +335,63 @@ def test_global_pooling(self, size, input_channels, batch_size,
+            op_type,
+            ["X"],
+            ["Y"],
+            kernels=[kernel] * dim,
```
Sure
